datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -402,25 +403,55 @@ def in_developer_mode(self):
         self._log_deprecation('in_developer_mode')
         return False
 
+    def find_typos_in_options(self, user_config, models_config, level):
```

Comment:
A minor nit: since the main purpose of this function is to log warnings, would it make sense to name it `log_typos_in_options` or something similar?